---
title: "Normalization RNA"
author: "memrekus"
date: "6/7/2022"
output: html_document
---


```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE)
```


#Load Libraries


```{r}

library("TCGAbiolinks")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
library("genefilter")
library("data.table")
library("dplyr")
library("tidyverse")
library("Rsamtools")
library("rtracklayer")
library("AnnotationHub")

```


#Download and prepare the raw data


```{r}

query_RNA <- GDCquery(
  
  project = "TCGA-SKCM",
                      
  data.category = "Transcriptome Profiling",
  
  experimental.strategy = "RNA-Seq",
  
  workflow.type = "STAR - Counts",
  
  data.type = "Gene Expression Quantification")

GDCdownload(query_RNA)

tcga_data<-GDCprepare(query_RNA)

rm(query_RNA)

```


#Create a cpm / log2 normalization 


```{r}

Normal_RNA<-assay(tcga_data)
  
Normal_RNA<-DGEList(counts = Normal_RNA, lib.size =colSums(Normal_RNA))

Normal_RNA<-cpm(Normal_RNA, normalized.lib.sizes = TRUE,
       log = TRUE, prior.count = 1)

```


#Formation of dataframe


```{r}

Normal_RNA<-as.data.frame(Normal_RNA)

Normal_RNA<-as.data.frame(t(Normal_RNA))

rownames(Normal_RNA)[1:473]<-substr((rownames(Normal_RNA)),1,16)

```


#Gene annotation


```{r}

gencodes<-tcga_data@rowRanges@elementMetadata@listData[["gene_name"]]

colnames(Normal_RNA)<- gencodes

```


#Save and remove 


```{r}

saveRDS(Normal_RNA,file = "Normal_RNA.Rds")

rm(tcga_data,gencodes)

```

